cg/src/utils/blockchain.js

```diff
@@ -176,6 +196,25 @@ async function listenerForRectificationEvent(callback) {
   });
 }
 
+async function listenerForRestrictionEvent(callback) {
+  const quorumContract = await getContract();
+  return await quorumContract.contract.events.Controller_SubjectDataRestricted((error, data) => {
```

Comment:
you can remove the `await` since you return the promise, because of `async`, the value is always a promise.

Reply:
We can't do that. All of those awaits are necessary.
Let me try to explain in simple terms, think of any blockchain call as a database query, it takes time but we need to wait until it finishes. In this case the listener won't be attached if we don't wait till it gets "set" in blockchain land

Reply:
Ok thanks. I will check if we have a ticket for the refactor.